source/neuropod/bindings/java/src/main/native/com_uber_neuropod_Neuropod.cc

```diff
@@ -0,0 +1,215 @@
+/* Copyright (c) 2020 UATC, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "com_uber_neuropod_Neuropod.h"
+
+#include "com_uber_neuropod_LibraryLoader.h"
+#include "jclass_register.h"
+#include "neuropod/core/generic_tensor.hh"
+#include "neuropod/neuropod.hh"
+#include "neuropod/tests/ope_overrides.hh"
+#include "utils.h"
+
+#include <exception>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include <jni.h>
+
+using namespace neuropod::jni;
+
+jobject toJavaTensorSpecList(JNIEnv *env, const std::vector<neuropod::TensorSpec> &specs);
+
+JNIEXPORT jlong JNICALL Java_com_uber_neuropod_Neuropod_nativeNew__Ljava_lang_String_2J(JNIEnv *env,
+                                                                                        jclass,
+                                                                                        jstring path,
+                                                                                        jlong   optHandle)
+{
+    try
+    {
+        neuropod::RuntimeOptions opts;
+        if (optHandle != 0)
+        {
+            opts = *reinterpret_cast<neuropod::RuntimeOptions *>(optHandle);
+        }
+        auto                convertedPath = toString(env, path);
+        neuropod::Neuropod *ret           = nullptr;
+        if (isTestMode)
+        {
+            ret = new neuropod::Neuropod(convertedPath, detail::ope_backend_location_overrides, opts);
+        }
+        else
+        {
+            ret = new neuropod::Neuropod(convertedPath, opts);
+        }
```

Comment:
Is there a cleaner way we can do this? Can we expose the `default_backend_overrides` argument to java?

`isTestMode` and `setTestMode` seem like a workaround

Reply:
ope_backend_location_overrides depends on current working path. I am afraid that the current working path in Java and C++ may be different in some cases.

Reply:
I also think it would be good to refactor code to not have isTestMode explicitly. And instead allow to pass/set ope_backend_location_overrides somehow. But at the same time I am ok if we proceed this way and get back to it later when get stable initial release.